src/options.c

```diff
@@ -351,7 +351,8 @@ static void print_usage(char *name)
 
 	printf(JOHN_USAGE, name);
 #if defined(HAVE_OPENCL) || defined(HAVE_CUDA)
-	printf("%s", JOHN_USAGE_GPU);
+	if (!none_opencl_device)
+		printf("%s", JOHN_USAGE_GPU);
```

Comment:
Perhaps the `none_opencl_device` should be in common-gpu.[ch] so it's there ever for a CUDA-only build. Or we need to have different logic, like
```C
#if defined(HAVE_OPENCL)
	if (!none_opencl_device)
		printf("%s", JOHN_USAGE_GPU);
#elif defined(HAVE_CUDA)
	printf("%s", JOHN_USAGE_GPU);
#endif
```